boot/bootutil/src/loader.c

```diff
@@ -773,6 +773,39 @@ boot_validate_slot(struct boot_loader_state *state, int slot,
         goto out;
     }
 
+#if BOOT_IMAGE_NUMBER > 1 && !defined(MCUBOOT_ENC_IMAGES) && defined(MCUBOOT_VERIFY_IMG_ADDRESS)
```

Comment:
The only thing I suggest is putting at least a comment in here specifying that this is Arm specific. The Kconfig catches that on Zephyr, but it could be confusing for someone on another platform to enable this.

This also catches situations, such as dual XIP, where the image linked at the wrong address gets loaded.

If this is a test build (with swap), the situation should actually recover after a reboot, and the images are reverted.

Reply:
> The only thing I suggest is putting at least a comment in here specifying that this is Arm specific. The Kconfig catches that on Zephyr, but it could be confusing for someone on another platform to enable this.

Yeah, I will add a comment.

> This also catches situations, such as dual XIP, where the image linked at the wrong address gets loaded.

I will see if I can add a note on this (at least in the kconfig help text)

> If this is a test build (with swap), the situation should actually recover after a reboot, and the images are reverted.

True. This is why I only `default y` if you don't perform swapping.

Reply:
This patch hurts the MCUboot hardware-agnostic architecture. Even if we eventually wrapped this in an `#ifdef ARM`, this opens the door to further shortcuts throughout the core library, which I believe would be greatly undesirable.

There should be a way for achieving the same result by properly creating a new generic interface in the abstraction layer.
All we need here is a method for identifying whether the image should run from slot 0 or 1, it should not be explicitly coupled to an architecture-specific detail.

Reply:
^^
Right. Then proper approach would be to define an interface which will be implemented by architecture dedicated source.